PWGHF/D2H/Tasks/taskFlowCharmHadrons.cxx

```diff
@@ -0,0 +1,339 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file taskFlowCharmHadrons.cxx
+/// \brief Analysis task for charm hadron flow
+///
+/// \author S. Politanò, INFN Torino, Italy
+
+#include "Framework/AnalysisTask.h"
+#include "Framework/HistogramRegistry.h"
+#include "Framework/runDataProcessing.h"
+
+#include "PWGHF/Core/HfHelper.h"
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+#include "PWGHF/DataModel/CandidateSelectionTables.h"
+
+#include "Common/DataModel/Qvectors.h"
+#include "Common/Core/EventPlaneHelper.h"
+
+using namespace o2;
+using namespace o2::aod;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+enum decayChannel { DplusToPiKPi = 0,
+                    DsToKKPi,
+                    DsToPiKK };
+
+struct taskFlowCharmHadrons {
+  Configurable<int> harmonic{"harmonic", 2, "harmonic number"};
+  Configurable<int> qvecDetector{"qvecDetector", 0, "Detector for Q vector estimation (FT0C: 0, FT0A: 1, FT0M: 2, FV0A: 3, TPC Pos: 4, TPC Neg: 5)"};
+  Configurable<int> centDetector{"centDetector", 0, "Detector for centrality estimation (V0A: 0, T0M: 1, T0A: 2, T0C: 3"};
+  Configurable<int> selectionFlag{"selectionFlag", 1, "Selection Flag for D"};
+  Configurable<bool> storeMl{"storeMl", false, "Flag to store ML score"};
+  Configurable<bool> saveEpResoHisto{"saveEpResoHisto", false, "Flag to save event plane resolution histogram"};
+  Configurable<int> classMl{"classMl", 0, "Index of the ML class to be stored"};
```

Comment:
Probably minor, but can you mention e.g. in the description which values this variable can have and to what it corresponds to?

Reply:
no clear suggestions from my side. Indeed I was wondering how much this might depend on the `MLResponse` organization, but from your reply I understand it's not easy to comment it here properly. Let's keep it in mind for the future.

Reply:
Ok, thanks! Maybe I could add the multiclass example for the D+ in the configurable definition, which I assume will be the most common case. But indeed, I don't think it will cover all the possible scenarios

Reply:
yes this can be already helpful, thanks!

Reply:
Hi @fcatalan92 , indeed we should find a way to organise the signal (prompt+ nonprompt) / background models in the vector storing the BDT information. For the moment, I will add a second BDT-related axis to store the bkg score, as you suggested. Thanks!